[Bug]: unar uncompress file fail & compile self fail.

### Problem description

When I use UNAR, I will fail to uncompress any 7Z or ZIP file, Some packs only release few part files.I tried to use the source code yourself and still failed.
```
[~]$ wget https://cdn.theunarchiver.com/downloads/TheUnarchiverSource.zip
[~]$ unar TheUnarchiverSource.zip
TheUnarchiverSource.zip: 2022-01-17 02:50:55.447 unar[5930:12970367422413591688] No local time zone specified.
2022-01-17 02:50:55.447 unar[5930:12970367422413591688] Using time zone with absolute offset 0.
Zip
  Extra/  (dir)... OK.
  Extra/lsar.1  (1670 B)... Failed! (Unknown error)
  Extra/lsar.1  (0 B, rsrc)... Failed! (Unknown error)
  Extra/lsar.bash_completion  (389 B)... Failed! (Unknown error)
  Extra/lsar.bash_completion  (0 B, rsrc)... Failed! (Unknown error)
  Extra/unar.1  (2596 B)... Failed! (Unknown error)
  Extra/unar.1  (0 B, rsrc)... Failed! (Unknown error)
  Extra/unar.bash_completion  (671 B)... Failed! (Unknown error)
  Extra/unar.bash_completion  (0 B, rsrc)... Failed! (Unknown error)
  Extra/  (dir, rsrc)... OK.
  License.txt  (27632 B)... Failed! (Unknown error)
  License.txt  (0 B, rsrc)... Failed! (Unknown error)
```
```
[~/unar]$ unar ../TheUnarchiverSource.zip ./
../TheUnarchiverSource.zip: 2022-01-17 03:07:14.942 unar[8927:12970367424673154184] No local time zone specified.
2022-01-17 03:07:14.943 unar[8927:12970367424673154184] Using time zone with absolute offset 0.
Zip                                                                              No files extracted.
```
```
[~/unar/XADMaster]$ make -f Makefile.linux                                       gcc -std=gnu99 -O2 -Wno-import -Wno-multichar -g -D_FILE_OFFSET_BITS=64 -isystem /usr/include/GNUstep -DGNUSTEP -DGNU_RUNTIME=1 -D_NATIVE_OBJC_EXCEPTIONS -fgnu-runtime -fexceptions -fobjc-exceptions -fconstant-string-class=NSConstantString   -MMD -MFBuild/XADTest2.d -MTBuild/XADTest2.o -c XADTest2.m -o Build/XADTest2.o    In file included from XADTest2.m:1:                                              In file included from ./XADArchiveParser.h:1:                                    In file included from /data/data/com.termux/files/usr/include/Foundation/Foundation.h:30:                                                                         /data/data/com.termux/files/usr/include/GNUstepBase/GSVersionMacros.h:294:6: error: "Your gnustep-base was configured for the objc-nonfragile-abi but you are not using it now."
#    error "Your gnustep-base was configured for the objc-nonfragile-abi but you are not using it now."                                                                ^                                                                           XADTest2.m:51:2: warning: format specifies type 'long long' but the argument has type 'uint64_t' (aka 'unsigned long') [-Wformat]                                         (uint64_t)[data length],crc,xor,xor);                                            ^~~~~~~~~~~~~~~~~~~~~~~                                                  1 warning and 1 error generated.                                                 make: *** [Makefile.linux:113: Build/XADTest2.o] Error 1                         [~/unar/XADMaster]$ gcc -v                                                       clang version 13.0.0                                                             Target: aarch64-unknown-linux-android24                                          Thread model: posix                                                              InstalledDir: /data/data/com.termux/files/usr/bin
```

### What steps will reproduce the bug?

use unar uncompress any 7Z or ZIP file or compile unar in termux.

### What is the expected behavior?

can not use this package or compile it.

### System information

termux-info:

```
[~/unar]$ termux-info                                                            Application version:
0.117                                                                            Packages CPU architecture:                                                       aarch64                                                                          Subscribed repositories:                                                         # sources.list                                                                   deb https://mirrors.tuna.tsinghua.edu.cn/termux/termux-packages-24 stable main
# game-repo (sources.list.d/game.list)                                           deb https://mirrors.tuna.tsinghua.edu.cn/termux/game-packages-24 games stable    # science-repo (sources.list.d/science.list)                                     deb https://mirrors.tuna.tsinghua.edu.cn/termux/science-packages-24 science stable                                                                                Updatable packages:
All packages up to date                                                          Android version:                                                                 11                                                                               Kernel build information:                                                        Linux 9pro 4.19.113-perf-g6bd56efa50c5 #1 SMP PREEMPT Mon Oct 25 22:16:18 CST 2021 aarch64 Android                                                                Device manufacturer:                                                             Xiaomi                                                                           Device model:                                                                    M2007J17C
```

--- a/packages/unar/build.sh
+++ b/packages/unar/build.sh
@@ -3,6 +3,7 @@ TERMUX_PKG_DESCRIPTION="Command line tools for archive and file unarchiving and 
 TERMUX_PKG_LICENSE="LGPL-2.1"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=()
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_VERSION+=(1.10.7)
 TERMUX_PKG_VERSION+=(1.1)
 TERMUX_PKG_SRCURL=(https://github.com/MacPaw/XADMaster/archive/v${TERMUX_PKG_VERSION}/XADMaster-${TERMUX_PKG_VERSION}.tar.gz

--- a/packages/unar/sys_time.c
+++ b/packages/unar/sys_time.c
@@ -34,21 +34,21 @@
 
 #pragma GCC visibility push(hidden)
 
-int timespec_from_timeval(struct timespec ts, const struct timeval tv) {
+int timespec_from_timeval(struct timespec* ts, const struct timeval tv) {
   // Whole seconds can just be copied.
-  ts.tv_sec = tv.tv_sec;
+  ts->tv_sec = tv.tv_sec;
 
   // But we might overflow when converting microseconds to nanoseconds.
   if (tv.tv_usec >= 1000000 || tv.tv_usec < 0) {
     return 0;
   }
-  ts.tv_nsec = tv.tv_usec * 1000;
+  ts->tv_nsec = tv.tv_usec * 1000;
   return 1;
 }
 
 static int futimesat(int fd, const char* path, const struct timeval tv[2], int flags) {
   struct timespec ts[2];
-  if (tv && (!timespec_from_timeval(ts[0], tv[0]) || !timespec_from_timeval(ts[1], tv[1]))) {
+  if (tv && (!timespec_from_timeval(&ts[0], tv[0]) || !timespec_from_timeval(&ts[1], tv[1]))) {
     errno = EINVAL;
     return -1;
   }

--- a/packages/uwsgi/build.sh
+++ b/packages/uwsgi/build.sh
@@ -3,6 +3,7 @@ TERMUX_PKG_DESCRIPTION="uWSGI application server container"
 TERMUX_PKG_LICENSE="GPL-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=2.0.19.1
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://github.com/unbit/uwsgi/archive/${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=bf17cdbb9bd8bcb7c1633e34d9d7308cb4cc19eb0ff2d61057f840c1ba1fc41b
 TERMUX_PKG_DEPENDS="libandroid-glob, libcap, libcrypt, libjansson, libuuid, libxml2, openssl, pcre, python"

--- a/packages/uwsgi/sys_time.c
+++ b/packages/uwsgi/sys_time.c
@@ -34,21 +34,21 @@
 
 #pragma GCC visibility push(hidden)
 
-int timespec_from_timeval(struct timespec ts, const struct timeval tv) {
+int timespec_from_timeval(struct timespec* ts, const struct timeval tv) {
   // Whole seconds can just be copied.
-  ts.tv_sec = tv.tv_sec;
+  ts->tv_sec = tv.tv_sec;
 
   // But we might overflow when converting microseconds to nanoseconds.
   if (tv.tv_usec >= 1000000 || tv.tv_usec < 0) {
     return 0;
   }
-  ts.tv_nsec = tv.tv_usec * 1000;
+  ts->tv_nsec = tv.tv_usec * 1000;
   return 1;
 }
 
 int futimes(int fd, const struct timeval tv[2]) {
   struct timespec ts[2];
-  if (tv && (!timespec_from_timeval(ts[0], tv[0]) || !timespec_from_timeval(ts[1], tv[1]))) {
+  if (tv && (!timespec_from_timeval(&ts[0], tv[0]) || !timespec_from_timeval(&ts[1], tv[1]))) {
     errno = EINVAL;
     return -1;
   }

